docs/DeveloperGuide.md

```diff
@@ -1,34 +1,579 @@
+7. Click `OK` to accept the default settings, if prompted. 
+8. Wait for the importing process to finish, and you are good to go!
+9. Verify the setup:
+    1. Run `Athena` and try a few commands such as `list` or `help`.
+    2. Run the JUnit tests to ensure they all pass.
+
+## **Design & implementation**
+
+### Architecture
+
+![Architecture Diagram](structures/ArchitectureDiagram.png)
+
+The ***Architecture Diagram*** given above explains the high-level design of ATHENA. A quick overview of each component is shown below.
+
+**`Main`** has one class called [`Athena`](https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/master/src/main/java/athena/Athena.java). It is responsible for:
+* At application launch: Initializes the components and connects them up with each other.
+* At application shut down: Shuts down the components.
+
+The rest of the application consists of these components:
+
+* [**`Ui`**](#athenaUi-component): The user interface of ATHENA.
+* [**`Logic`**](#logic-component): Parses user input and executes commands.
+* [**`TaskList`**](#tasklist-component): The list that stores the user's tasks.
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+The *Sequence Diagram* below shows how these components interact with each other for the scenario where the user issues a command.
+
+*work in progress*
+
+The sections below give more details for each component.
+
+### UI component
+
+![UiStructure](https://raw.githubusercontent.com/alstontham/tp/DG-UG/docs/structures/UiStructure.png)
+
+**API** :
+[`Ui.java`](https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/master/src/main/java/athena/Ui.java)
+
+1. `Ui` consists of `AthenaUi` and `ColorText`
+2. `AthenaUi` implements `Ui` interface and outputs messages that the user sees
+3. `ColorText` applies relevant colors to certain output messages
+4. `AthenaUi` also prints out error messages tied to `Command Exception` that get thrown by `Athena`
+5. `LogicManager` requires `AthenaUi` to execute user commands
+
+### Logic component
+
+![Structure of the Logic Component](structures/LogicStructure.png)
+
+**API** :
+[`Logic.java`](https://github.com/AY2021S1-CS2113T-W12-2/tp/blob/master/src/main/java/athena/logic/Logic.java)
+
+1. `Logic` consists of `LogicManager`, which calls the `Parser` class to parse the user command.
+2. This results in a `Command` object being created, which is then executed by the `LogicManager`.
+3. The command execution can affect the `TaskList` (e.g. Adding a task).
+4. The result of the command execution will also call the `Ui` to print the respective messages for each command.
+
+The following sequence diagram illustrates how the `LogicManager` works:
+
+![LogicManagerSequenceDiagram](sequenceDiagrams/LogicManager%20diagram.png)
+
+The following sequence diagram illustrates how the `Parser` works:
+
+![ParserSequenceDiagram](sequenceDiagrams/Parser%20diagram.png)
```

Comment:
Should the footbox be removed instead? and the function name should be aligned with the arrow?
<img width="564" alt="Screenshot 2020-10-30 at 14 11 31" src="https://user-images.githubusercontent.com/56800018/97667482-ef8a7580-1aba-11eb-88e4-6eee9d49a0e0.png">